src/Spreadsheet.tsx

```diff
@@ -30,89 +25,99 @@ import * as Actions from "./actions";
-  DataViewer: Types.DataViewer<CellType, Value>,
-  DataEditor: Types.DataEditor<CellType, Value>,
-  onKeyDown?: (event: SyntheticKeyboardEvent<HTMLElement>) => void,
-  getValue: Types.getValue<CellType, Value>,
-  getBindingsForCell: Types.getBindingsForCell<CellType>,
-  store: Store
-|};
-
-type Handlers = {|
-  cut: () => void,
-  copy: () => void,
-  paste: () => void,
-  setDragging: boolean => void,
-  onKeyDownAction: (SyntheticKeyboardEvent<HTMLElement>) => void,
-  onKeyPress: (SyntheticKeyboardEvent<HTMLElement>) => void,
-  onDragStart: () => void,
-  onDragEnd: () => void
-|};
-
-type State = {|
-  rows: number,
-  columns: number,
-  mode: Types.Mode
-|};
+export type Props<CellType, Value> = {
+  data: Matrix.Matrix<CellType>;
+  columnLabels?: string[];
+  ColumnIndicator?: React.FC<ColumnIndicatorProps>;
+  rowLabels?: string[];
+  RowIndicator?: React.FC<RowIndicatorProps>;
+  hideRowIndicators?: boolean;
+  hideColumnIndicators?: boolean;
+  Table: React.FC<TableProps>;
+  Row: React.FC<RowProps>;
+  Cell: React.FC<CellProps<CellType, Value>>;
+  DataViewer: Types.DataViewer<CellType, Value>;
+  DataEditor: Types.DataEditor<CellType, Value>;
+  onKeyDown?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+  getValue: Types.getValue<CellType, Value>;
+  getBindingsForCell: Types.getBindingsForCell<CellType>;
+  store: Store<any>;
+};
+
+type Handlers = {
+  cut?: () => void;
+  copy?: () => void;
+  paste?: () => void;
+  setDragging?: (_: boolean) => void;
+  onKeyDownAction?: (_: React.SyntheticEvent<HTMLElement>) => void;
+  onKeyPress?: (_: React.SyntheticEvent<HTMLElement>) => void;
+  onDragStart?: () => void;
+  onDragEnd?: () => void;
+};
+
+type State = {
+  rows: number;
+  columns: number;
+  mode: Types.Mode;
+};
 
 type ColumnIndicatorProps = {
-  column: number,
-  label?: Node | null
+  column: number;
+  label?: string | null;
 };
 
-const DefaultColumnIndicator = ({ column, label }: ColumnIndicatorProps) =>
+const DefaultColumnIndicator: React.FC<ColumnIndicatorProps> = ({
+  column,
+  label
+}: ColumnIndicatorProps) =>
   label !== undefined ? (
     <th>{label}</th>
   ) : (
     <th>{columnIndexToLabel(column)}</th>
   );
 
 type RowIndicatorProps = {
-  row: number,
-  label?: Node | null
+  row: number;
+  label?: string | null;
 };
 
-const DefaultRowIndicator = ({ row, label }: RowIndicatorProps) =>
+const DefaultRowIndicator: React.FC<RowIndicatorProps> = ({
+  row,
+  label
+}: RowIndicatorProps) =>
   label !== undefined ? <th>{label}</th> : <th>{row + 1}</th>;
 
-class Spreadsheet<CellType, Value> extends PureComponent<{|
-  ...$Diff<
-    Props<CellType, Value>,
-    {|
-      data: Matrix.Matrix<CellType>
-    |}
-  >,
-  ...State,
-  ...Handlers
-|}> {
+interface SpreadsheetProps<CellType, Value>
+  extends Props<CellType, Value>,
+    State,
+    Handlers {
+  data: Matrix.Matrix<CellType>;
+}
+
+class Spreadsheet<CellType, Value> extends React.Component<
```

Comment:
Unclear why this was switched to Component instead of PureComponent.

> React.PureComponent is similar to React.Component. The difference between them is that React.Component doesn’t implement shouldComponentUpdate(), but React.PureComponent implements it with a shallow prop and state comparison.
```suggestion
class Spreadsheet<CellType, Value> extends React.PureComponent<
```